test/e2e-ansible/e2e_ansible_olm_test.go

```diff
@@ -26,22 +26,14 @@ var _ = Describe("Integrating ansible Projects with OLM", func() {
 		const operatorVersion = "0.0.1"
 
 		It("should generate and run a valid OLM bundle and packagemanifests", func() {
-			By("building the bundle")
-			err := tc.Make("bundle", "IMG="+tc.ImageName)
-			Expect(err).NotTo(HaveOccurred())
-
-			By("building the operator bundle image")
-			err = tc.Make("bundle-build", "BUNDLE_IMG="+tc.BundleImageName)
-			Expect(err).NotTo(HaveOccurred())
-
 			if tc.IsRunningOnKind() {
 				By("loading the bundle image into Kind cluster")
-				err = tc.LoadImageToKindClusterWithName(tc.BundleImageName)
+				err := tc.LoadImageToKindClusterWithName(tc.BundleImageName)
 				Expect(err).NotTo(HaveOccurred())
 			}
 
 			By("adding the 'packagemanifests' rule to the Makefile")
-			err = tc.AddPackagemanifestsTarget()
+			err := tc.AddPackagemanifestsTarget()
 			Expect(err).NotTo(HaveOccurred())
```

Comment:
I'm not a fan of separating OLM integration setup code into `BeforeEach` and test specs for the sake of encapsulation/refactoring. Can you call `IntegrateProjectWithOLM` in the test specs? It may be a bit of duplicated code but it is clearer.

Reply:
We cannot to do that because we are not replacing all files with the dev image. Detailed info can be found in  https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/127

Reply:
Are you saying to call the `IntegrateProjectWithOLM` in the olm tests. Am I right? 

I would really like to that, however, it is only possible if my first suggestion made in https://github.com/operator-framework/operator-sdk/issues/3922 was accepted. See that because we NOT replacing all files with the dev image we need to re-run the bundle which has the cons described in the issue: 

- The solution is not idempotent because if we remove the commands from the suite test then we will no longer be using the dev images for scorecard which means that we can introduce silent  bugs that will get only after the release
- We cannot let all bundle operations only in the test where it is required which means that we are hurting concepts as cohesion 
- And then we are probably spending more CI effort to re-run all instead of just do the replace. 

So, I would happy to address your suggestion in a follow-up PR with the solution proposed  in https://github.com/operator-framework/operator-sdk/issues/3922.

Reply:
>The solution is not idempotent because if we remove the commands from the suite test then we will no longer be using the dev images for scorecard which means that we can introduce silent bugs that will get only after the release

Is that really true? I'm just suggesting you keep all the OLM-related setup code in one place.

>We cannot let all bundle operations only in the test where it is required which means that we are hurting concepts as cohesion

Are you saying you want to run bundle builds for all tests even if they don't make use of bundles?

>And then we are probably spending more CI effort to re-run all instead of just do the replace.

I'm not sure I understand what you're saying. I wouldn't expect more code to be run with my suggested changes, just where it is run from.

Reply:
> Are you saying you want to run bundle builds for all tests even if they don't make use of bundles?

Yes. Otherwise, it will have manifests for the scorecard with the released image tag because instead of we replace all files/manifests we are just replacing the specific ones, see: https://github.com/operator-framework/operator-sdk/blob/master/test/e2e-go/e2e_go_suite_test.go#L70-L76. The solution made in https://com/operator-framework/operator-sdk/pull/3909 mainly perform the replace in 2 places and run the make bundle to ensure that all will get updated. 

As described in https://com/operator-framework/operator-sdk/pull/3909#discussion_r490586393 and in issue #3922. IMO we should:
- In the suite test replace all manifest to use the dev image ( config/bundle/dockerfile ) 
- The bundle commands would only be called in the before func of the olm_test as you suggested. 

However, it shows out of the scope of this PR and it is exactly what I have been trying to reach a consensus with #3922.

Reply:
@estroz, 
 
Now, I hope that my motivations for #3922 and are more clear. So, wdyt about we agree in I will push a PR with your suggestion and the fix for it described in #3922 as a follow up of this one and then it could make easier we move forward. Could we agree with that?

Reply:
I don't think citing an issue (#3922) that is still under discussion, and doesn't look like something we'll address as a reason given the comments, is a good reason to make these changes, or at least not accept my proposed changes. If that is the reason, I recommend closing this PR because the solution to that issue hasn't been resolved.

Reply:
hI @estroz, 

The goal of this PR has centralized the duplications. **It is ONLY a cleanup.**

You are doing a suggestion that cannot be done. The root cause for your suggestion cannot be in place can be found in https://com/operator-framework/operator-sdk/pull/3909#discussion_r490586393 and in #3922. Indeed your suggestion and comments here are my motivations for #3922 which is not part of the scope of this PR. 

So, I do not see why it should not be merged. In this way, I'd like to move forward here and we still discussing your suggestion in #3922. Could we agree with that?

Reply:
PS.: Just to clarifies, the change that you asked cannot be done because of how it is merged/done already in master via the solution proposed in  https://com/operator-framework/operator-sdk/pull/3909 and not because of any change in this PR.

Reply:
it has been updated a while this thread is too big and confusing. 
So, let's start from scratch.